docs/troubleshooting.md

```diff
@@ -0,0 +1,74 @@
+
+It is ok to learn by experiment and backward engineering. Kubernetes eco-system to provide that flexibility but only needs bit of tinkering around to get the ASO running. Let me take you through a scenario:
+
+Lets assume that that as an end user Tats povided wrong service principal which was incorrect for the ASO to be correctly set-up.
+
+**Sample scenario**: Lets take a user sceanrio example where `user-A` mistakenly used an out-dated or wrong service principal secret for `azure-service-operator` which will result in unsuccessful result, how can we rectify now:
+
+* ~~ToDo: I plan to add specific user-case scenario here: (possibly with screenshot)~~ (Do we need screenshot mockup here?) 
+
+* So at this point the issue happend in step-4 mentioned [here](https://operatorhub.io/operator/azure-service-operator), essentially all user need to do is re-new or refresh their service-principal credentials.
+    * Re-new the service principal, helpful command mentioned ins step-4 [here](https://operatorhub.io/operator/azure-service-operator). 
+
+    * Then after re-generating that user need to make sure that the only steps they need to manually run are step-5 onwards mentioned here: https://operatorhub.io/operator/azure-service-operator , But we can simplify a little more here as most of these command will be available. use command : `az account show` to get `AZURE_TENANT_ID` is the `Tenant ID`, `AZURE_SUBSCRIPTION_ID`, `AZURE_CLIENT_ID`. (This is just indicative, you can follow how to get these variable from Step-5 here https://operatorhub.io/operator/azure-service-operator)
+
+        * Decoding the cloud env var for the users: https://docs.microsoft.com/en-us/azure/storage/common/storage-powershell-independent-clouds#get-endpoint-using-get-azenvironment 
+
+    * create `azureoperatorsettings.yaml` file from step-6 mentioend here: https://operatorhub.io/operator/azure-service-operator replace the placeholder values into the respective field and do `kubectl apply -f <location of your azureoperatorsettings.yaml fil>`
+
+    * User should be able to see `azureoperator-controller-manager-***` pods running in there cluster. helpful command to see the pod: `kubectl get pods -n operators` 
+
+## I have already installed `azureserviceoperator` but unsure what service principal ceret is in use?
+this command might be able to help you for some recon for your secret locally. `kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'` [SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
```

Comment:
```suggestion
## I have already installed `azureserviceoperator` but I'm not sure which service principal secret is in use?
This command might be able to help you see your secret locally. 
```kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'```
[SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
```